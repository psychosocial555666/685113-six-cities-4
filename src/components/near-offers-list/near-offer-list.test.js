import React from "react";
import renderer from "react-test-renderer";
import NearOffersList from "./near-offers-list.jsx";

const offers = [
  {
    title: `Wood and stone place`,
    img: `src`,
    price: 80,
    rating: 5,
    type: `Apartment`,
    isItPremium: true,
    id: 1,
    cityName: `Paris`,
    images: [
      `src`,
      `src`,
      `src`,
      `src`,
      `src`,
      `src`,
    ],
    description: `Some text`,
    bedrooms: 2,
    guests: 3,
    appliances: [
      `Wi-Fi`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    owner: {
      avatar: `src`,
      name: `Death`,
      isSuper: true,
    },
    coords: [52.39874984984841, 4.82456445558843],
    reviews: [
      {
        reviewId: 14,
        reviewName: `Ozzy`,
        reviewAvatar: `src`,
        reviewGrade: 4,
        reviewText: `Id minim labore ut velit sit velit.Magna deserunt reprehenderit consequat elit cupidatat proident nostrud amet minim nulla.`,
        reviewTime: `May 8, 2016`,
      },
      {
        reviewId: 22,
        reviewName: `Mick`,
        reviewAvatar: `src`,
        reviewGrade: 2,
        reviewText: `Incididunt fugiat non aliqua eu nisi.Id Lorem cillum non voluptate nulla id fugiat Lorem exercitation irure ullamco enim veniam ullamco.`,
        reviewTime: `June 5, 2018`,
      },
    ],
    nearOffers: [1]
  },
];

const onCardMouseEnter = jest.fn();
const onCardMouseLeave = jest.fn();
const onHeaderClick = jest.fn();

it(`Should NearOffersList render correctly`, () => {
  const tree = renderer
      .create(<NearOffersList offers={offers}
        onHeaderClick={onHeaderClick}
        onCardMouseLeave={onCardMouseLeave}
        onCardMouseEnter={onCardMouseEnter}/>, {
        createNodeMock: () => {
          return document.createElement(`div`);
        }
      }
      )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
