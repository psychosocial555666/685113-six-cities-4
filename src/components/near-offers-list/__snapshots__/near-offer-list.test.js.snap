// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NearOffersList Should NearOffersList render correctly 1`] = `
<section
  className="near-places places"
>
  <h2
    className="near-places__title"
  >
    Other places in the neighbourhood
  </h2>
  <div
    className="near-places__list places__list"
  >
    <article
      className="near-places__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="near-places__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="src"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              80
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            Wood and stone place
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
  </div>
</section>
`;
